### ISSUE TITLE
[Frameworks] XPU/XCCL all_gather + torch.xpu.empty_cache leaks device-global memory

### ISSUE THREAD

**Issue by nscottnichols** [argonne-lcf/AuroraBugTracking#143]
### Point of Contact

Nathan Nichols

### Contact Details

MLSL-4397

### Vendor/ALCF/other tickets/IDs

https://github.com/intel/torch-xpu-ops/issues/3744

### Reproducer Path

See issue
source/reproducers_frameworks/allgather_hidden_temp_memleak

### Status

Open

### Details

If PyTorch does an XCCL allgather using XPU device memory and then the tensor is deleted and `torch.xpu.empty_cache()` is called, PyTorch thinks it has released the memory, but the device still does not get all of it back. The missing memory accumulates every iteration until the GPU reports out-of-memory.

This is not from ordinary PyTorch allocator caching. PyTorch's own counters go back down, but `torch.xpu.mem_get_info()` shows the actual device free memory keeps dropping.

A short list of things checked:
  - If we reuse the same allgather buffers every iteration, memory stays flat.
  - If `empty_cache()` forces new device addresses every iteration, memory leaks.
  - If we allocate and free memory without an XCCL collective, it does not leak.
  - If we do the same all-gather using a persistent flat output tensor, it does not leak.
  - If we use list-style `dist.all_gather`, PyTorch internally creates a hidden temporary flattened output buffer. That hidden buffer leaks after `empty_cache()`.

To reproduce ([see here for more details](https://github.com/argonne-lcf/frameworks-sdk/tree/main/tests/single_node/functionality/torch-xpu-ops/allgather_hidden_temp_memleak)):
```
module load frameworks
mkdir -p allgather_hidden_temp_memleak
curl -L https://github.com/argonne-lcf/frameworks-sdk/archive/refs/heads/main.tar.gz |
  tar -xz -C allgather_hidden_temp_memleak --strip-components=6 \
  --wildcards '*/tests/single_node/functionality/torch-xpu-ops/allgather_hidden_temp_memleak/*'
./allgather_hidden_temp_memleak/run_repros.sh
```

For the list-style `dist.all_gather`, the leak is very explicit. In the reproducer the hidden temp should be 0.373 GiB. The measured memory loss is exactly 0.373 GiB per iteration. Manually creating the same temp and using it in all-gather reproduces the same leak. Creating the same temp without using it in all-gather does not leak (sometimes there is a very small leak, but maybe that is from garbage collection?).

Workaround:
Avoid calling `torch.xpu.empty_cache()` in training loops that use XCCL collectives. Reusing collective buffers is stable.

Intel says it's from oneCCL (MLSL-4397)

### Is this a priority/blocking bug?

- [x] Priority
